File: lailahailaahhah/src/main/java/com/sami/rippel/lailahailaahhah/MainActivity.java
package com.sami.rippel.lailahailaahhah;

import java.io.IOException;

import android.app.Activity;
import android.app.WallpaperManager;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.SharedPreferences.Editor;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.RadioGroup;
import android.widget.RadioGroup.OnCheckedChangeListener;
import android.widget.Toast;

public class MainActivity extends Activity {
    private static long back_pressed;
    private Editor prefsEditor;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        SharedPreferences mPref = getSharedPreferences(Constants.PREFERENCESNAME,
                Context.MODE_PRIVATE);
        prefsEditor = mPref.edit();
        findViewById(R.id.buttonSetWallpaper).setOnClickListener(new OnClickListener() {
            public void onClick(View v) {
                ClearCurrentWallpaper();
                try {
                    Intent intent = new Intent(
                            WallpaperManager.ACTION_CHANGE_LIVE_WALLPAPER);
                    intent.putExtra(
                            WallpaperManager.EXTRA_LIVE_WALLPAPER_COMPONENT,
                            new ComponentName(getApplicationContext(),
                                    IslamicWallpaper.class));
                    startActivity(intent);
                } catch (Exception e) {
                    Intent intent = new Intent();
                    intent.setAction(WallpaperManager.ACTION_LIVE_WALLPAPER_CHOOSER);
                    startActivity(intent);
                    Toast.makeText(getApplicationContext(), R.string.txt1,
                            Toast.LENGTH_SHORT).show();
                }
            }
        });
        findViewById(R.id.btnOtherApp).setOnClickListener(new OnClickListener() {

            public void onClick(View v) {
                Intent intent = new Intent(Intent.ACTION_VIEW);
                intent.setData(Uri.parse("market://search?q=pub:SFAXDROID"));
                startActivity(intent);
            }

        });

        findViewById(R.id.imageViewPub).setOnClickListener(new OnClickListener() {

            public void onClick(View v) {
                Intent intent = new Intent(Intent.ACTION_VIEW);
                intent.setData(Uri
                        .parse("market://details?id=com.sami.rippel.allah"));
                startActivity(intent);
            }
        });

        RadioGroup radioWallpaper = (RadioGroup) findViewById(R.id.radioWallpaper);
        String qualitypref = mPref.getString(Constants.ChangeImageKey, "none");

        if (qualitypref.equalsIgnoreCase("Allah 1")) {
            radioWallpaper.check(R.id.wallp1);
        } else if (qualitypref.equalsIgnoreCase("Allah 2")) {
            radioWallpaper.check(R.id.wallp2);
        } else if (qualitypref.equalsIgnoreCase("Allah 3")) {
            radioWallpaper.check(R.id.wallp3);
        } else {
            radioWallpaper.check(R.id.wallp1);
            prefsEditor.putString(Constants.ChangeImageKey, "Allah 1");
            prefsEditor.apply();
        }

        radioWallpaper
                .setOnCheckedChangeListener(new OnCheckedChangeListener() {
                    @Override
                    public void onCheckedChanged(RadioGroup group, int checkedId) {
                        switch (checkedId) {
                            case R.id.wallp1:
                                prefsEditor.putString(Constants.ChangeImageKey,
                                        "Allah 1");
                                prefsEditor.commit();
                                break;
                            case R.id.wallp2:
                                prefsEditor.putString(Constants.ChangeImageKey,
                                        "Allah 2");
                                prefsEditor.commit();
                                break;
                            case R.id.wallp3:
                                prefsEditor.putString(Constants.ChangeImageKey,
                                        "Allah 3");
                                prefsEditor.commit();
                                break;
                        }
                    }
                });
    }

    public void ClearCurrentWallpaper() {
        WallpaperManager myWallpaperManager = WallpaperManager
                .getInstance(MainActivity.this);
        try {
            myWallpaperManager.clear();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onBackPressed() {
        if (back_pressed + 2000 > System.currentTimeMillis())
            super.onBackPressed();
        else {
            RateUs.app_launched(this);
            Toast.makeText(getBaseContext(), R.string.txtrate6,
                    Toast.LENGTH_SHORT).show();
        }
        back_pressed = System.currentTimeMillis();
    }
}

File: lailahailaahhah/src/main/java/com/sami/rippel/lailahailaahhah/MainActivity.kt
package com.sami.rippel.lailahailaahhah

import android.content.Context
import android.os.Bundle
import com.sfaxdroid.base.BaseActivity
import com.sfaxdroid.base.Utils
import com.sfaxdroid.engine.Constants
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        buttonSetWallpaper.setOnClickListener {
            ClearCurrentWallpaper()
            Utils.openLiveWallpaper<WallpaperEngine>(this)
        }

        imageViewPub.setOnClickListener {
            Utils.openPub(this)
        }

        initRadioButton()

    }

    private fun initRadioButton() {
        val sharedPref = getSharedPreferences(
            Constants.PREFERENCE_NAME,
            Context.MODE_PRIVATE
        )

        val editor = sharedPref.edit()

        when (sharedPref.getString(Constants.CHANGE_IMAGE_KEY, "")) {
            Constants.CHANGE_IMAGE_VALUE_ONE -> {
                radioWallpaper.check(R.id.wallp1)
            }
            Constants.CHANGE_IMAGE_VALUE_TWO -> {
                radioWallpaper.check(R.id.wallp2)
            }
            Constants.CHANGE_IMAGE_VALUE_THREE -> {
                radioWallpaper.check(R.id.wallp3)
            }
            else -> {
                radioWallpaper.check(R.id.wallp1)
                editor.putString(
                    Constants.CHANGE_IMAGE_KEY,
                    Constants.CHANGE_IMAGE_VALUE_ONE
                )
                editor.apply()
            }
        }
        radioWallpaper
            .setOnCheckedChangeListener { _, checkedId ->
                when (checkedId) {
                    R.id.wallp1 -> {
                        editor.putString(
                            Constants.CHANGE_IMAGE_KEY,
                            Constants.CHANGE_IMAGE_VALUE_ONE
                        )
                        editor.commit()
                    }
                    R.id.wallp2 -> {
                        editor.putString(
                            Constants.CHANGE_IMAGE_KEY,
                            Constants.CHANGE_IMAGE_VALUE_TWO
                        )
                        editor.commit()
                    }
                    R.id.wallp3 -> {
                        editor.putString(
                            Constants.CHANGE_IMAGE_KEY,
                            Constants.CHANGE_IMAGE_VALUE_THREE
                        )
                        editor.commit()
                    }
                }
            }
    }
}